src/libraries/System.Runtime.InteropServices/gen/ComInterfaceGenerator/ComMethodInfo.cs

```diff
@@ -123,7 +124,12 @@ private static DiagnosticOr<(ComMethodInfo, IMethodSymbol)> CalculateMethodInfo(
             {
                 attributeInfos.Add(AttributeInfo.From(attr));
             }
-            var comMethodInfo = new ComMethodInfo(comMethodDeclaringSyntax, method.Name, attributeInfos.MoveToImmutable().ToSequenceEqual());
+            var shadowsBaseMethod = false;
+            if (comMethodDeclaringSyntax.Modifiers.Any(SyntaxKind.NewKeyword))
+            {
+                shadowsBaseMethod = true;
+            }
+            var comMethodInfo = new ComMethodInfo(comMethodDeclaringSyntax, method.Name, attributeInfos.MoveToImmutable().ToSequenceEqual(), shadowsBaseMethod);
```

Comment:
Do we construct `ComMethodInfo` instance in other places?

Reply:
I ask because this seems precisely like the sort of thing that should be in a constructor rather than just hanging out in some method.

Reply:
This is the only place we construct this type. I agree it makes sense to limit where this could be constructed, but we should probably do that for all the XInfo and XContext types we have and should be a separate change. I'll file an issue to make sure I don't forget.

Reply:
I don't think that is needed to be commented on. My point here is that we should consider this thought when applying the `record` pattern. I'm still not convinced we use it properly or as intended and I think we've applied it a bit prematurely into this codebase.

Reply:
Yeah, I think we only used record early when the code was changing significantly since it gave us free equality, and I agree it's time to change it back to a regular class and write our own equality.